src/client/datascience/commands/activeEditorContext.ts

```diff
@@ -100,24 +115,28 @@ export class ActiveEditorContextService implements IExtensionSingleActivationSer
     private onDidChangeActiveInteractiveWindow(e?: IInteractiveWindow) {
         this.interactiveContext.set(!!e).ignoreErrors();
         this.updateMergedContexts();
+        this.updateContextOfActiveInteractiveWindowKernel();
     }
```

Comment:
You can use vscode built in context variable `notebookCellExecuting`
That'll be easier

Reply:
Should we adopt that for the notebook editor as well then?

Reply:
I just tried it out, that context key only works when you set focus to the notebook cell that is executing. I don't think that will work for our use case.

Reply:
Try this `notebookHasRunningCell`

Reply:
That wouldn't work well too, e.g. if a notebook and IW are open side by side, running a cell in a notebook will cause the interrupt button in the IW toolbar to light up, and vice versa. Also in the IW case, since the IW may not actually be active (focus may be in the Python file), that would cause issues as well. The current implementation still suffers from this limitation https://github.com/microsoft/vscode-jupyter/issues/7487 for SxS IWs, but at least allows us to ensure that the IW toolbar is enabled correctly based on the active #%% file or active IW, and to ensure that IWs' and notebooks' run state doesn't affect the toolbar button enablement for each other.

Reply:
yuck, agreed.